src/lib/secrets_helper.ts

```diff
@@ -133,20 +82,46 @@ const formatSecrets = (result: OneGraphSecretsResponse | undefined) => {
   return newSecrets
 }
 
-type OneGraphPayload = { authlifyToken: string | undefined }
+interface RequestHeaders {
+  get(name: string): string | null
+}
+
+interface IncomingMessageHeaders {
+  [key: string]: string
+}
+
+interface HasHeaders {
+  headers: RequestHeaders | IncomingMessageHeaders
+}
 
-export type HandlerEventWithOneGraph = HandlerEvent & OneGraphPayload
+const hasRequestStyleHeaders = function (headers: RequestHeaders | IncomingMessageHeaders): headers is RequestHeaders {
+  return (headers as RequestHeaders).get !== undefined && typeof headers.get === 'function'
+}
+
+// This function accepts null for backwards compatibility with version < 0.11.1,
+// where getSecrets did not require an event
+const graphTokenFromEvent = function (event: HasHeaders | null): string | null {
+  if (!event) {
+    return null
+  }
+
+  const { headers } = event
+  // Check if object first in case there is a header with key `get`
+  if (TOKEN_HEADER in headers) {
+    return headers[TOKEN_HEADER]
```

Comment:
You'll need to narrow this, as it could be an array where you'd want to return the first value.